src/urql/atomWithMutation.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  TypedDocumentNode,
+  OperationContext,
+  OperationResult,
+} from '@urql/core'
+import { atom } from 'jotai'
+import type { Atom, Getter } from 'jotai'
+import { getClient } from './clientAtom'
+
+type MutationAction<Data, Variables extends object> = {
+  variables?: Variables
+  context?: Partial<OperationContext>
+  callback?: (result: OperationResult<Data, Variables>) => void
+}
+
+export function atomWithMutation<Data, Variables extends object>(
+  createQuery: (get: Getter) => TypedDocumentNode<Data, Variables>
+): Atom<OperationResult<Data, Variables>> {
+  const operationResultAtom = atom<
+    OperationResult<Data, Variables> | Promise<OperationResult<Data, Variables>>
+  >(
+    new Promise<OperationResult<Data, Variables>>(() => {}) // infinite pending
+  )
+  const queryResultAtom = atom(
+    (get) => get(operationResultAtom),
+    (get, set, action: MutationAction<Data, Variables>) => {
+      const client = getClient(get, set)
+      const query = createQuery(get)
+      client
+        .mutation(query, action.variables, action.context)
+        .toPromise()
+        .then((result) => {
+          set(operationResultAtom, result)
+          action.callback?.(result)
+        })
+        .catch(() => {
+          // TODO error handling
+        })
+    }
+  )
```

Comment:
atomWithMutation is much simpler than atomWithQuery as it doesn't have onMount nor subscription.
It might be tricky to use because`useAtom` will suspend. So, either creating another atom or `useUpdateAtom` is necessary.

Reply:
I think that's acceptable though. I'd expect that people would have to use `client.mutation` directly for their side-effects half the time, since atoms are really good for reactive values but in this case we're dealing with mutations, which are less reactive and more like an imperative side-effect